### PR TITLE
Fix S1172 FP: Unused parameter cannot be removed

### LINKED ISSUE #8371: Fix S1172 FP: Unused parameter cannot be removed
### Description

Rule S1172 suggests removing unused parameters from a method that is used as an event handler. Therefore the unused parameters cannot be removed without breaking the code.

### Repro steps

The code below produces two warnings for unused parameters. While the parameters are unused, they cannot be removed:

```c#
#nullable enable

using System.Reflection;

namespace SonarParameterBug;

internal static class Program
{
    public static void Main()
    {
        Console.WriteLine(typeof(AssemblyLoader));
    }
}

internal static class AssemblyLoader
{
    static AssemblyLoader()
    {
        AppDomain.CurrentDomain.AssemblyResolve += AssemblyResolver.LoadAnyVersion;
    }

    private static class AssemblyResolver
    {
        // S1172: Remove this unused method parameter 'sender'.
        // S1172: Remove this unused method parameter 'args'.
        public static Assembly? LoadAnyVersion(object? sender, ResolveEventArgs args)
        {
            throw new NotSupportedException();
        }
    }
}
```

### Expected behavior

No warnings.

### Actual behavior

Incorrect warnings. The parameters cannot be removed.

### Known workarounds

Moving the method `LoadAnyVersion` outside of the nested class makes the warnings go away.

### Related information

* C#/VB.NET Plugins version: N/A
* Visual Studio version: Visual Studio Community 2022 v17.8.0
* MSBuild / dotnet version: .NET 6
* SonarScanner for .NET version: 9.12.0.78982
* Operating System: Windows 11

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Fixes #8371

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MethodParameterUnused.RoslynCfg.Fixed.cs [hunk 1]
```diff
@@ -213,7 +215,7 @@ public static class Extensions
             return a + b;
         }
 
-        private static int AddedLength(this string s, int a ) //Fixed
+        private static int AddedLength(this string s, int a) //Fixed
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MethodParameterUnused.RoslynCfg.Fixed.cs:218]: That is what `SyntaxRemoveOptions.AddElasticMarker` is for.

File: analyzers/src/SonarAnalyzer.CSharp/Rules/MethodParameterUnusedCodeFix.cs [hunk 2]
```diff
@@ -46,7 +46,7 @@ protected override Task RegisterCodeFixesAsync(SyntaxNode root, SonarCodeFixCont
                 {
                     var newRoot = root.RemoveNode(
                         parameter,
-                        SyntaxRemoveOptions.KeepLeadingTrivia);
+                        SyntaxRemoveOptions.KeepLeadingTrivia | SyntaxRemoveOptions.AddElasticMarker);
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/MethodParameterUnusedCodeFix.cs:49]: Unrelated minor fix. See for details

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MethodParameterUnused.RoslynCfg.cs [hunk 3]
```diff
@@ -105,7 +106,8 @@ void LocalFunctionLocalLifetime(int localArgLocalLifetime)
                 }
             }
 
-            void DoSomething() { }
+            void DoSomething()
+            { }
```
Comment (sebastien-marichal) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MethodParameterUnused.RoslynCfg.cs:110]: Any reason for this change or just automatic formating?

**Review by sebastien-marichal — APPROVED** [review]
LGTM!
I just had one question out of curiosity.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MethodParameterUnused.RoslynCfg.cs (hunk 3 above)
Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MethodParameterUnused.RoslynCfg.cs:110]: VS fixes some formatting errors when you copy/paste the expected result. I changed the code, so it is now copy/paste friendly. (I fixed the "elastic marker" problem for the same reason).
